pkg/kstatus/polling/clusterreader/caching_reader.go

```diff
@@ -239,7 +239,12 @@ func (c *CachingClusterReader) Sync(ctx context.Context) error {
 		list.SetGroupVersionKind(mapping.GroupVersionKind)
 		err = c.reader.List(ctx, &list, listOptions...)
 		if err != nil {
-			// We continue even if there is an error. Whenever any pollers
+			// If the context was cancelled, we just stop the work and return
+			// the error.
+			if err == context.Canceled || err == context.DeadlineExceeded {
```

Comment:
Are we interested in supporting wrapped errors? i.e. `errors.Is(err, context.Canceled)` instead of equality. https://go.dev/blog/go1.13-errors

Reply:
Yeah, I agree that is better. I have updated the PR.